Formula/planck.rb

```diff
@@ -21,8 +21,40 @@ class Planck < Formula
   depends_on "icu4c"
   depends_on "libzip"
 
+  uses_from_macos "vim" => :build # for xxd
+  uses_from_macos "curl"
+  uses_from_macos "libffi"
+
+  on_linux do
+    depends_on "gcc"
+    depends_on "glib"
+    depends_on "pcre"
+    depends_on "webkitgtk"
+  end
+
+  fails_with gcc: "5"
+
+  # Apply upstream commit to fix issue with GNU sed.  Remove with next release.
+  patch do
+    url "https://github.com/planck-repl/planck/commit/f1f21bf9eb4cfca6312ff0117f75d3b38164b43d.patch?full_index=1"
+    sha256 "787ddf6fb1cfea1d70fa18a6f7b292579ea1ffbf1f437f1f775e3330d2b8d36c"
+  end
+
   def install
     ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix
+
+    unless OS.mac?
+      ENV.prepend_path "PATH", Formula["openjdk"].opt_bin unless OS.mac?
+
+      # The webkitgtk pkg-config .pc file includes the API version in its name (ex. javascriptcore-4.1.pc).
+      # We extract this from the filename programmatically and store it in javascriptcore_api_version
+      # and make sure planck-c/CMakeLists.txt is updated accordingly.
+      javascriptcore_pc_file = (Formula["webkitgtk"].share/"pkgconfig").glob("javascriptcoregtk-*.pc").first.to_s
+      javascriptcore_api_version = File.basename(javascriptcore_pc_file,
+                                                 File.extname(javascriptcore_pc_file)).split("-")[1]
```

Comment:
```suggestion
      javascriptcore_pc_file = (Formula["webkitgtk"].share/"pkgconfig").glob("javascriptcoregtk-*.pc").first
      javascriptcore_api_version = javascriptcore_pc_file.basename(".pc").to_s.split("-").second
```

Reply:
I think we're missing a type conversion here because the call to basename is failing:
```
Error: An exception occurred within a child process: NoMethodError: undefined method `basename' for nil:NilClass
```

Reply:
`javascriptcore_pc_file == nil` so that would mean glob found nothing.

`pkgconfig` files are usually installed in `lib` so `Formula["webkitgtk"].share` doesn't look right.